### PR TITLE
New Api draft

### PR DESCRIPTION
**Pull request by MrAlexEsisteGia**
Still a draft and I need some confirmations but I hope it's not too wrong
the ? means I'm not too sure
and the ?tbd means I need to research more to make an educated guess

still learning javascript on the side so sorry for any dumb mistakes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,3 +7,53 @@
 [![Try it on gitpod](https://img.shields.io/badge/try-on%20gitpod-brightgreen.svg)](https://gitpod.io/#https://github.com/PrismarineJS/prismarine-rng)
 
 Collection of utilities to work with minecraft RNG.
+
+## API
+
+## LCG
+?tbd
```
Comment (Karang) [inline README.md]: Linear Congruential Generator https://en.wikipedia.org/wiki/Linear_congruential_generator
The type of random number generator used by Java.

File: README.md [hunk 2]
```diff
@@ -7,3 +7,53 @@
...
+
+## Random
+?tbd
```
Comment (Karang) [inline README.md]: An implementation of Java's Random https://docs.oracle.com/javase/8/docs/api/java/util/Random.html

File: README.md [hunk 3]
```diff
@@ -7,3 +7,53 @@
...
+
+## crackPlayerSeed
+### crackPlayerSeed.crack(bits)
+Returns the decrypted player seed
+* `Returns` - Seed number
+* `bits` - Encrypted seed string?
```
Comment (Karang) [inline README.md]: tbh, this is function is a bit useless without an example. I need to add it.
bits expect and array of 20 values recorded from throwing objects on the ground and measuring their velocity.

File: README.md [hunk 4]
```diff
@@ -7,3 +7,53 @@
...
+
+
+## enchantments
+### enchantments.findEnchantment(playerSeed, item, power, matching)
+* `Returns` - Object or null ?tbd
+* `playerSeed` - Number ?tbd
+* `item` - Item object ?tbd
+* `power` - ?tbd
+* `matching` - Matching function that returns a bool ?tbd
```
Comment (Karang) [inline README.md:36]: Maybe it'll be easier to understand with an example:
```js
const playerSeed = 0x484b921be1c6n
const item = new Item(mcData.itemsByName.diamond_sword.id, 0)
console.log(Enchantments.findEnchantment(playerSeed, item, 15, (list) => {
  return list.find(e => e.enchant.name === 'sharpness' && e.level === 4) &&
         list.find(e => e.enchant.name === 'looting' && e.level === 3) &&
         list.find(e => e.enchant.name === 'sweeping' && e.level === 3) &&
         list.find(e => e.enchant.name === 'unbreaking' && e.level === 3)
}))
```
item is the object you want to enchant
power is the number of bookshelf arround the table
matching is a function that return true if the enchant is what you are looking for

File: README.md [hunk 5]
```diff
@@ -7,3 +7,53 @@
...
+
+### enchantments.getEnchantmentList(xpseed, item, slot, level)
+* `Returns` - Array ?tbd
+* `xpseed` - ?tbd
```
Comment (Karang) [inline README.md]: the seed of the rng for the next enchant

File: README.md [hunk 6]
```diff
@@ -7,3 +7,53 @@
...
+## enchantments
+### enchantments.findEnchantment(playerSeed, item, power, matching)
+* `Returns` - Object or null ?tbd
+* `playerSeed` - Number ?tbd
+* `item` - Item object ?tbd
+* `power` - ?tbd
+* `matching` - Matching function that returns a bool ?tbd
+
+### enchantments.getEnchantmentList(xpseed, item, slot, level)
+* `Returns` - Array ?tbd
```
Comment (Karang) [inline README.md:39]: the list of enchantment you'll get for the slot

File: README.md [hunk 7]
```diff
@@ -7,3 +7,53 @@
...
+* `xpseed` - ?tbd
+* `item` - Item object ?tbd
+* `slot` - ?tbd
+* `level` - Number ?tbd
+
+### enchantments.getEnchantmentCost(rand, slot, power, item)
```
Comment (Karang) [inline README.md:45]: the cost (xp) of a slot

File: README.md [hunk 8]
```diff
@@ -7,3 +7,53 @@
...
+* `Returns` - Numeber ?tbd
+* `rand` - ?tbd
+* `slot` - ?tbd
+* `power` - ?tbd
+* `item` - ?tbd
+
+### enchantments.getEnchantmentPower(pos, world)
```
Comment (Karang) [inline README.md:52]: count bookshelves

File: README.md [hunk 9]
```diff
@@ -7,3 +7,53 @@
...
+* `Returns` - Power number ?tbd
+* `pos` - Vec3 ?tbd
+* `world` - ?tbd
+
+## createWorldGenerator
+### createWorldGenerator(mcData, worldSeed, dimension)
+Returns a promise
+* `Returns` - Promise that resolves to ?tbd
```
Comment (Karang) [inline README.md:60]: a world generator

**Review by Karang — COMMENTED** [review]
precised a few points

File: README.md (hunk 3 above)
Comment (Karang) [reply README.md]: Added some example on how to use it https://github.com/PrismarineJS/prismarine-rng/blob/master/examples/enchant.js
Would be nice to add a notice in the doc that this is highly experimental and only a library to help building seed cracking tool, not a tool itself. I think for someone to be able to truely using it, he must know the basics of seed cracking already.

File: README.md (hunk 5 above)
Comment (MrAlexEsisteGia) [reply README.md]: im assuming its the description for xpseed

**Comment by MrAlexEsisteGia** [comment]
still a draft but im going to continue later
thanks for going into detail

**Comment by rom1504** [comment]
reopen to continue it

**Pull request closed without merging**